src/services/lineaStateBridgeEventListener.js

```diff
@@ -0,0 +1,76 @@
+import { ethers } from "ethers";
+
+export const lineaStateBridgeEventListener = async ({
+  ethereumEndpoint,
+  lineaStateBridgeAddress,
+  l1MessageServiceAddress,
+}) => {
+  let provider;
+  let reconnectAttempts = 0;
+  const MAX_RECONNECT_ATTEMPTS = 10;
+  const RECONNECT_DELAY = 5000; // 5 seconds delay between reconnection attempts
+
+  const initializeProvider = () => {
+    provider = new ethers.providers.JsonRpcProvider(ethereumEndpoint);
+
+    // Listen for provider events
+    provider.on('connect', () => {
+      console.log('Connected to the Ethereum network.');
+      reconnectAttempts = 0; // Reset attempts on successful connection
+    });
+
+    provider.on('error', (error) => {
+      console.error('Network error:', error);
+    });
+
+    provider.on('disconnect', (error) => {
+      console.error('Disconnected from the Ethereum network:', error);
+      handleReconnection();
+    });
+  };
+
+  // Function to handle reconnection
+  const handleReconnection = () => {
+    if (reconnectAttempts < MAX_RECONNECT_ATTEMPTS) {
+      reconnectAttempts++;
+      console.log(`Attempting to reconnect... (${reconnectAttempts}/${MAX_RECONNECT_ATTEMPTS})`);
+      setTimeout(() => {
+        initializeProvider();
+        listenForEvents(); // Reattach event listeners after reconnection
+      }, RECONNECT_DELAY);
+    } else {
+      console.error('Max reconnection attempts reached. Exiting...');
+      process.exit(1); // Exit process after max attempts
+    }
+  };
+
```

Comment:
I made a quick research and found out that Ethers V6 handle disconnects itself and don't have `connect` and `disconnect` events on a provider, so I guess we can remove it. Moreover, this code will result in an error.